cmd/api/main.go

```diff
@@ -177,5 +214,27 @@ func main() {
 	} else {
 		log.Fatal(http.ListenAndServeTLS(":"+configFile.REST_PORT, path.Join(configFile.TLS_PATH, "tls.crt"), path.Join(configFile.TLS_PATH, "tls.key"), mux))
 	}
+}
+
+func determineAuth(ctx context.Context) (context.Context, error) {
+
+	tokenString, err := grpc_auth.AuthFromMD(ctx, "bearer")
+	if err != nil {
+		ctxzap.AddFields(ctx,
+			zap.String("grpc.user", "unauthenticated_request"),
+		)
+		return ctx, nil
+	}
 
+	token, _, _ := new(jwt.Parser).ParseUnverified(tokenString, jwt.MapClaims{})
```

Comment:
Two items:

- Error is a potential return here. Should we raise an error, treat it as an unauthenticated request, or something else?
- Please comment why we are using `ParseUnverified`. The godoc has an explict warning ("do not use unless you know what you are doing")

Reply:
1. I will update the code to handle the error returned and log it but I think we should return the error to the calling method as it should stop the rest of the code from processing, WDYT ?
2. 
3. We are using `ParseUnverifed` to just take the name of the sub from the token. we don't verify it. that's still left up to the existing code to verify. Do you think we should verify it here also ?

Reply:
> I will update the code to handle the error returned and log it but I think we should return the error to the calling method as it should stop the rest of the code from processing, WDYT ?

I do have a concern that since this is an interceptor, an error causes the request to fail. Perhaps we set `grpc.user` to `unknown` in the context?

Reply:
Updated to include those points.